indexer/config.go

```diff
@@ -121,18 +118,17 @@ type Config struct {
 func NewConfig(ctx *cli.Context) (Config, error) {
 	cfg := Config{
 		/* Required Flags */
-		BuildEnv:                ctx.GlobalString(flags.BuildEnvFlag.Name),
-		EthNetworkName:          ctx.GlobalString(flags.EthNetworkNameFlag.Name),
-		ChainID:                 ctx.GlobalInt64(flags.ChainIDFlag.Name),
-		L1EthRpc:                ctx.GlobalString(flags.L1EthRPCFlag.Name),
-		L2EthRpc:                ctx.GlobalString(flags.L2EthRPCFlag.Name),
-		L1AddressManagerAddress: ctx.GlobalString(flags.L1AddressManagerAddressFlag.Name),
-		L2GenesisBlockHash:      ctx.GlobalString(flags.L2GenesisBlockHashFlag.Name),
-		DBHost:                  ctx.GlobalString(flags.DBHostFlag.Name),
-		DBPort:                  ctx.GlobalUint64(flags.DBPortFlag.Name),
-		DBUser:                  ctx.GlobalString(flags.DBUserFlag.Name),
-		DBPassword:              ctx.GlobalString(flags.DBPasswordFlag.Name),
-		DBName:                  ctx.GlobalString(flags.DBNameFlag.Name),
+		BuildEnv:           ctx.GlobalString(flags.BuildEnvFlag.Name),
+		EthNetworkName:     ctx.GlobalString(flags.EthNetworkNameFlag.Name),
+		ChainID:            ctx.GlobalInt64(flags.ChainIDFlag.Name),
+		L1EthRpc:           ctx.GlobalString(flags.L1EthRPCFlag.Name),
+		L2EthRpc:           ctx.GlobalString(flags.L2EthRPCFlag.Name),
+		L2GenesisBlockHash: ctx.GlobalString(flags.L2GenesisBlockHashFlag.Name),
```

Comment:
What do you think about removing the L2 genesis block hash and instead use an L1 chain id and L2 chain id flag for validation that its talking to the correct chains?

Reply:
+1 on this. Can be handled separately but it's easier to pass in a chain ID than a genesis hash.

Reply:
OK, think it was being used to partially index the chain (flag should be called `L2StartBlockHashFlag`), but I don't see it being much useful outside testing. Will remove.

Reply:
Will handle this in a new PR.